design-proposals/cluster-orch-integrating-k8s-into-emt.md

```diff
@@ -0,0 +1,313 @@
+      schema:
+        openAPIV3Schema:
+          type: boolean
+          default: false
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: cluster-on-emt
+spec:
+  topology:
+    ...
+    variables:
+    - name: airGapped
+      value: true
+```
+
+For all other cases, including K3s on Ubuntu machine and RKE2 on EMT/Ubuntu machine, this configuration will remain false.
+
+### Workflow Change
+
+Clusters for a specific host can be created in two ways:
+
+**1. Automatically during host registration**: This new option is designed for users with simple use cases who want Kubernetes to be available with a pre-defined template, reducing user interactions during onboarding and cluster creation.
+
+**2. Manually through a direct request to the Cluster Manager**: This method allows users to create clusters on hosts that have already been onboarded (or provisioned, depending on the finalized design). It is particularly useful in scenarios where host onboarding and workload creation occur independently, often managed by different users with distinct roles or at different times. For example, an Onboarding Manager might register a pool of hosts with only the auto-onboard option enabled (and potentially auto-provision, based on the final design). Later, Edge Operators or Users can select an operating system and create clusters using an OS profile, Kubernetes flavor, and template that best align with specific workload requirements. This approach provides flexibility and adaptability, making it well-suited for cloud-like edge usage.
+
+Before diving into workflow changes, it is essential to understand the coupling introduced by this design change between the following components:
+
+- **EMT Image Version**: Specifies the operating system and embedded software, including the K3s version, on the edge device.
+- **Cluster Template**: Defines the configuration for Kubernetes cluster creation, including the Kubernetes flavor, version, and custom control plane settings.
+
+For instance, attempting to create a K3s cluster using a template for K3s v1.32 on an EMT machine embedding K3s v1.30 may result in compatibility issues. Such mismatches can occur when Edge Orchestration undergoes multiple upgrades, leading to a mix of cluster templates and EMT machines with varying versions. While this issue is specific to the K3s-on-EMT scenario, it is critical to address it effectively in the user workflow, as this represents the most common use case for EMF.
+
+#### Automatic Cluster Creation
+
+The first option introduces a streamlined process for users seeking simplicity or bulk registration of hosts, or both. Both the Web UI and CLI will offer a new toggle for `Create Cluster Automatically` (potentially combined with `Provision Automatically`, pending final design decisions) during registration. This option is only valid when `Provision Automatically` is enabled. When enabled, it automatically creates a single-node cluster for each host.
```

Comment:
For some deployments (like minimal OXM deployment) we won't deploy cluster orchestration. In that case we need to configure UI accordingly to not show the new toggle for CO. CC: @teone